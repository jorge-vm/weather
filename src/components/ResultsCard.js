
const ResultsCard = ({ weather, city }) => {
  const { temp, humidity, temp_max:max, temp_min:min } = weather;
  return (
    <div className="card-group">
    <div className="card">
      <div className="card-body">
        <h5 className="card-title">Weather in {city}</h5>
        <p className="card-text">Temperature {temp}°</p>
      </div>
    </div>
    <div className="card">
      <div className="card-body">
        <h5 className="card-title">Min and Max</h5>
        <p className="card-text">Max: {max}°</p>
        <p className="card-text"> Min: {min}°</p>
      </div>
    </div>
    <div className="card">
      <div className="card-body">
        <h5 className="card-title">Humidity</h5>
        <p className="card-text">{humidity}%</p>
      </div>
    </div>
  </div>
  );
};

export default ResultsCard;
