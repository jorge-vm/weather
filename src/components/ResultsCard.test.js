import { render } from "@testing-library/react";
import ResultsCard from "./ResultsCard";

test("renders ResultsCard", () => {
  const weather = { temp: 90, humidity: 45, temp_max: 43, temp_min: 45 };
  const city = "Miami";

  const { container } = render(<ResultsCard weather={weather} city={city} />);
  expect(container.firstChild).toMatchSnapshot();
});
