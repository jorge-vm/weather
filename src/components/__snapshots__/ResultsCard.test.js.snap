// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResultsCard 1`] = `
<div
  class="card-group"
>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Weather in 
        Miami
      </h5>
      <p
        class="card-text"
      >
        Temperature 
        90
        °
      </p>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Min and Max
      </h5>
      <p
        class="card-text"
      >
        Max: 
        43
        °
      </p>
      <p
        class="card-text"
      >
         Min: 
        45
        °
      </p>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Humidity
      </h5>
      <p
        class="card-text"
      >
        45
        %
      </p>
    </div>
  </div>
</div>
`;
